rcl/src/rcl/timer.c

```diff
@@ -402,7 +402,14 @@ rcl_timer_reset(rcl_timer_t * timer)
   }
   int64_t period = rcutils_atomic_load_uint64_t(&timer->impl->period);
   rcutils_atomic_store(&timer->impl->next_call_time, now + period);
+  bool is_canceled;
+  is_canceled = rcutils_atomic_load_bool(&timer->impl->canceled);
   rcutils_atomic_store(&timer->impl->canceled, false);
+  if (is_canceled) {
```

Comment:
why just triggering it when it was cancelled?
doesn't make sense to always trigger it?

Reply:
Yes, you're right. Although the timer is not canceled timer, gc MUST always be triggered in
order to rcl_wait->rmw_wait MUST reset timeout. I applied your comment.